### PR TITLE
chore: update to compatibility with latest version of vm2

### PR DESCRIPTION
**Pull request by joonazan**
Had to adjust the integration a bit because of vm2 interface changes. Adjusted the amount of gas passed to the VM.

With this version vm2 only fails TransientStorage tests. Those opcodes are unimplemented.

### CODE REVIEW

**Comment by joonazan** [comment]
Looks like vm2 will need to be public first. Also, I have realized I have to make more changes to the VM's interface to be able to integrate into core, so I think it makes sense to postpone this a bit.

**Review by hedgar2017 — APPROVED** [review]

**Review by hedgar2017 — CHANGES_REQUESTED** [review]
The tester is public now, so it can only work with public repos.
Can you FOS vm2?

**Review by hedgar2017 — APPROVED** [review]

**Review by hedgar2017 — CHANGES_REQUESTED** [review]
Could you squash your commits and force-reapply them? Unfortunately the base was changed after merging v1.5.0, and I'd like not to resolve all these conflicts manually.

**Comment by joonazan** [comment]
@hedgar2017 I just removed all your commits that were not related to vm2 and the conflicts went away. This is ok because they are in main anyway, right?

**Comment by hedgar2017** [comment]
> @hedgar2017 I just removed all your commits that were not related to vm2 and the conflicts went away. This is ok because they are in main anyway, right?

Yes, I believe the history got broken is some way after merging `v1.5.0`.
Unfortunately I still can see the conflicts in this PR.

File: Cargo.toml [hunk 1]
```diff
@@ -7,3 +7,6 @@ members = [
     "fuzzer"
 ]
 resolver = "2"
+
+[patch."https://github.com/matter-labs/vm2"]
+vm2 = {path = "../vm2"}
```
Comment (hedgar2017) [inline Cargo.toml:12]: Probably that's why it fails.

**Review by hedgar2017 — CHANGES_REQUESTED** [review]

**Review by hedgar2017 — APPROVED** [review]

**Pull request merged**